vm/devices/storage/disk_nvme/nvme_driver/src/store_fence.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// UNSAFETY: Running raw assembly/intrinsic.
+#![allow(unsafe_code)]
+
+/// Emit a store fence to flush the processor's store buffer.
+pub(crate) fn store_fence() {
+    // The compiler emits `mfence` on x86_64 and `dmb ...` on aarch64
+    // for `std::atomic::fence()` hence the intrinsics and the assembly below.
+    #[cfg(target_arch = "x86_64")] // xtask-fmt allow-target-arch cpu-intrinsic
+    {
+        // SAFETY: this instruction has no safety requirements.
+        unsafe { std::arch::x86_64::_mm_sfence() };
```

Comment:
Could we move all of this to the safe_x86_intrinsics crate and a new safe_aarch64_intrinsics crate? Then we can keep the forbid(unsafe) on this crate.

Reply:
Can we just call it `safe_intrinsics` instead of having two crates?